docs-ui/components/button.stories.js

```diff
@@ -216,24 +216,20 @@ storiesOf('UI|Buttons', module)
         </Section>
 
         <Section>
-          <h3>Merged Buttons</h3>
-          <ButtonBar merged>
-            <Button priority="primary" className="active">
-              Left Button
-            </Button>
-            <Button>Right Button</Button>
+          <h3>Merged Buttons with "active" button</h3>
+          <ButtonBar active="left" merged>
+            <Button id="left">Left Button</Button>
+            <Button id="right">Right Button</Button>
           </ButtonBar>
         </Section>
 
         <Section>
-          <h3>Multiple Merged Buttons</h3>
-          <ButtonBar merged>
-            <Button>First Button</Button>
-            <Button priority="primary" className="active">
-              Second Button
-            </Button>
-            <Button>Third Button</Button>
-            <Button>Fourth Button</Button>
+          <h3>Multiple Merged Buttons with "active" button</h3>
+          <ButtonBar active={2} merged>
```

Comment:
Would this interface be simpler if it was `activeIndex={number}` and we counted child indexes? Coupling to a prop on the child element could make typings tricky for class/function components that don't support `id` props.

Reply:
Hmm typings is a good point. Since this is a `ButtonBar`, is it safe to assume we'll only ever use `Button` as children?

My concern with using an index is that the data is not as closely coupled with the children being rendered. In the above example where `id` is `left|right`, our logic would probably be something like `const activeIndex = [left, right].indexOf(params.position)` and if you were to change the order of the `Buttons` it could cause regressions.

Reply:
I would hope we generally are grouping buttons too. Do we ever group elements like dropdown buttons with standard buttons?

If we stick with id, could we use string names and not allow numeric ids? Having strings removes the possibility of confusing the ids as order.

Reply:
DropdownButtons are just Buttons, so they should still work.

Hmm, now that I think of it with this implementation, `id` gets removed and never passed to `<Button>`, which is bad if we need the html attribute `id` -- and I think like you said, if we used a different prop, it would cause typing issues.

Reply:
Would having to use a component from `buttonBar` the module be gross? e.g.

```
import ButtonBar, {ButtonBarChild} from 'app/components/buttonBar'

<ButtonBar>
<ButtonBarChild someId="foo" />
<ButtonBarChild someId="bar" />
</ButtonBar>
```

Reply:
Having a separate child helps with overloading `id`, but makes adapting existing code a bit harder as more code needs to change. I looked at Material-UI and they don't have a good model for this active state that we could steal.

Reply:
Pushed an update using a diff prop - it requires basically a placeholder prop on `Button`